docs/src/pages/api-reference/server.mdx

```diff
@@ -149,6 +149,84 @@ accessible here
 The UploadThing API Helper, for use ON YOUR SERVER. It's basically just a REST
 API but better
 
+### `uploadFiles`
+
+> Added in `v5.3`
+
+`uploadFiles` let's you upload files directly from your server, without having
+to define file routes. When uploadthing files from the server, no presigned URLs
+are used, your files are uploaded via our servers.
+
+This can be useful for uploading files from sever actions, or your own API
+routes.
+
+```tsx
+import { utapi } from "uploadthing/server";
+
+async function uploadFiles(formData: FormData) {
+  "use server";
+  const files = fd.getAll("files");
+  const response = await utapi.uploadFiles(files);
+  //    ^? { key: string, url: string }[]
+}
+
+function MyForm() {
+  return (
+    <form action={uploadFiles}>
+      <input name="files" type="file" multiple />
+      <button type="submit">Upload</button>
+    </form>
+  );
+}
+```
+
+| Arguments | Type                       | Required | Description                       |
+| :-------- | :------------------------- | :------- | :-------------------------------- |
+| files     | FileEsque \|\| FileEsque[] | Yes      | The file(s) you want to upload    |
+| metadata  | Json                       | No       | Metadata to attach to the file(s) |
+
+```ts
+// Edge / Node < 20 friendly File interface
+interface FileEsque extends Blob {
+  name: string;
+}
+```
+
+### `uploadFilesFromUrl`
+
+> Added in `v5.3`
+
+`uploadFilesFromUrl` is similar to `uploadFiles` as it let's you upload files
+directly from your server, without having to define file routes.
+
+When uploading files from URL, the file is first downloaded on **your** server,
```

Comment:
may want to put this in a callout block -- since it could be costly for the user.